staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/apis/webhookadmission/types.go

```diff
@@ -16,14 +16,67 @@ limitations under the License.
 
 package webhookadmission
 
-import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // WebhookAdmission provides configuration for the webhook admission controller.
 type WebhookAdmission struct {
-	metav1.TypeMeta
+	metav1.TypeMeta `json:",inline"`
 
 	// KubeConfigFile is the path to the kubeconfig file.
-	KubeConfigFile string
+	KubeConfigFile string `json:"kubeConfigFile"`
+
+	// ResourceFilters associate a key with a list of rules that match resources.
+	// +optional
+	Metrics WebhookAdmissionSpecMetrics `json:"metrics,omitempty"`
+}
+
+type WebhookAdmissionSpecMetrics struct {
+	// SLODuration specifies options when observing the admission webhook SLO duration metrics.
+	// +optional
+	SLODuration WebhookAdmissionSpecMetricsSLODuration `json:"sloDuration,omitempty"`
+}
+
+type WebhookAdmissionSpecMetricsSLODuration struct {
+	// IncludeResourceLabelsFor is a list of rules which when matched by a resource will add
+	// resource specific labels to the observed metric.
+	// +optional
+	IncludeResourceLabelsFor []Rule `json:"includeResourceLabelsFor,omitempty"`
+}
+
+// Rule matches resources matching the specified lists of groups versions and resources.
+type Rule struct {
+
+	// Scope of this rule. "Cluster", "Namespaced", or "*". The default, "*", matches all scopes.
+	// +optional
+	Scope *Scope `json:"scope,omitempty"`
+
+	// Groups match API groups. "*" entry matches all.
+	Groups []string `json:"groups,omitempty"`
+
+	// Versions match API version. "*" entry matches all.
+	Versions []string `json:"versions,omitempty"`
+
+	// Resources match API resource names. "*" entry matches all.
+	Resources []string `json:"resources,omitempty"`
+
+	// Namespaces match API resource namespaces. "*" entry matches all. Empty string matches cluster scope.
+	// +optional
+	Namespaces []string `json:"namespaces,omitempty"`
 }
```

Comment:
I don't understand why we match the scope and the namespace here. The original issue comes from the fact that the `resource` label is unbounded. As such wouldn't it be better to just have rules to allow certain resources? I am not sure about the use case of allowing resources at the scope/ns level, I would rather see it at a global level directly.

Reply:
> The original issue comes from the fact that the resource label is unbounded. As such wouldn't it be better to just have rules to allow certain resources?

I concur. Updated.

Reply:
> So my suggestion would be to remove the operation label for now.

Done

Reply:
> Also, if you feel that we should rename the metrics and drop the admission part, I am fine with it +1

There is already a corresponding `apiserver_crd_webhook_conversion_duration_seconds` metric. So I think it is ok to leave this as admission webhook only.